fields/acf-city_selector-v5.php

```diff
@@ -128,17 +129,33 @@ function render_field( $field ) {
 					<?php if ( $field['show_labels'] == 1 ) { ?>
 						<span class="acf-input-header"><?php esc_html_e( 'Select a province/state', 'acf-city-selector' ); ?></span>
 					<?php } ?>
-					<label for="stateCode" class="screen-reader-text"></label>
-					<select name="<?php echo $field['name']; ?>[stateCode]" id="stateCode" class="countrySelect">
+					<label for="<?php echo $field['name']; ?>[stateCode]" class="screen-reader-text"></label>
+					<select name="<?php echo $field['name']; ?>[stateCode]" id="<?php echo str_replace("]", "", str_replace('[', "", $field['name'])); ?>stateCode" class="countrySelect">
+						<?php $states = get_states( $countrycode ); ?>
+						<?php if ($states): ?>
+							<?php foreach ($states as $state_key => $state_value): ?>
+								<option value="<?php echo "$countrycode-$state_key"; ?>" <?php echo $state_key == $field['value']['stateCode'] ? 'selected' : ''; ?>>
+									<?php echo "$state_value"; ?>
+								</option>
+							<?php endforeach ?>
+						<?php endif ?>
 					</select>
 				</div>
 
 				<div class="dropdown-box cs-cities">
 					<?php if ( $field['show_labels'] == 1 ) { ?>
 						<span class="acf-input-header"><?php esc_html_e( 'Select a city', 'acf-city-selector' ); ?></span>
 					<?php } ?>
-					<label for="cityName" class="screen-reader-text"></label>
-					<select name="<?php echo $field['name']; ?>[cityName]" id="cityName" class="countrySelect">
+					<?php $cities = get_cities( $countrycode, $field['value']['stateCode'] ); ?>
```

Comment:
Undefined variable: countrycode

Reply:
```
if ( isset( $field[ 'value' ][ 'countryCode' ] ) ) {
$countrycode = $field[ 'value' ][ 'countryCode' ];
}
```
okay, replace it to ...
```
$countrycode = null;
if ( isset( $field[ 'value' ][ 'countryCode' ] ) ) {
$countrycode = $field[ 'value' ][ 'countryCode' ];
}
```

Reply:
Why did you place PHP code in an element which is generated by javascript ?

Reply:
I'm not sure what you wanna ask ... the element is called through ajax and the ajax requests dynamic file ....

Reply:
You added option values for states and cities, where they are loaded dynamically. See [here](https://github.com/Beee4life/acf-city-selector/blob/master/assets/js/city-selector.js#L98) for states and [here](https://github.com/Beee4life/acf-city-selector/blob/master/assets/js/city-selector.js#L124) for cities. So I don't understand (yet) why you code this in when it's generated ?